ext/hal/cmsis/DSP/Kconfig

```diff
@@ -0,0 +1,509 @@
+# Copyright (c) 2019 Stephanos Ioannidis <root@stephanos.io>
+# SPDX-License-Identifier: Apache-2.0
+
+comment "Components"
+
+config CMSIS_DSP_BASICMATH
+	bool "Basic Math Functions"
+	default y
+
+config CMSIS_DSP_COMPLEXMATH
+	bool "Complex Math Functions"
+	imply CMSIS_DSP_FASTMATH
```

Comment:
Just a note, but if `CMSIS_DSP_FASTMATH` starts out disabled in the initial configuration, then turning on `CMSIS_DSP_COMPLEXMATH` in menuconfig won't enable it.

Has to do with [this gotcha](https://docs.zephyrproject.org/latest/guides/kconfig/tips.html#stuck-symbols-in-menuconfig-and-guiconfig).

Reply:
> Just a note, but if `CMSIS_DSP_FASTMATH` starts out disabled in the initial configuration, then turning on `CMSIS_DSP_COMPLEXMATH` in menuconfig won't enable it.

That is indeed the case.

`CMSIS_DSP_FASTMATH` is "implied" instead of "selected" here because, depending on the "complex math" function being used, `CMSIS_DSP_FASTMATH` may not be required (and may be de-selected).

> Has to do with [this gotcha](https://docs.zephyrproject.org/latest/guides/kconfig/tips.html#stuck-symbols-in-menuconfig-and-guiconfig).

I reviewed the workarounds listed in the link above and none of them seem to provide an elegant solution.

With that being the case, is there a way to display a note/warning to the user if an implied symbol is not selected?

Reply:
> With that being the case, is there a way to display a note/warning to the user if an implied symbol is not selected?

Besides checking the output, a warning could be added in `menuconfig`/`guiconfig` like this:

```
comment "Warning: You probably want foo enabled when bar is enabled"
	depends on BAR && !FOO
```

Was mostly of a note at least. Not sure how problematic it is in this case.